test/end2end_test.go

```diff
@@ -3901,6 +3901,36 @@ func (s) TestClientRequestBodyErrorCloseAfterLength(t *testing.T) {
 	}
 }
 
+func testClientInvalidStreamID(t *testing.T, e env) {
+	te := newTest(t, e)
```

Comment:
Do we need to create a `test` object here? The test struct does a lot of other things which are not significant for this test. 

Can we just create a gRPC server (calling `grpc.NewServer()`) and use that instead?